curvine-server/src/master/fs/master_actor.rs

```diff
@@ -33,31 +37,73 @@ impl MasterActor {
         executor: Arc<GroupExecutor>,
     ) -> Self {
         Self {
-            fs,
+            fs: Arc::new(fs),
             master_monitor,
             executor,
         }
     }
 
-    pub fn start(&self) {
+    pub fn start(&mut self) {
         info!("start master actor");
         Self::start_heartbeat_checker(
             self.fs.clone(),
             self.master_monitor.clone(),
             self.executor.clone(),
         )
         .unwrap();
+
+        if let Err(e) = self.start_ttl_scheduler() {
+            error!("Failed to start TTL scheduler: {}", e);
+        }
+    }
+    pub fn start_ttl_scheduler(&mut self) -> CommonResult<()> {
+        info!("Starting TTL scheduler with new architecture");
+        let ttl_manager = InodeTtlManager::create(self.fs.clone(), self.fs.conf.clone())
+            .map_err(|e| format!("Failed to create TTL system: {}", e))?;
+        let ttl_manager_arc = Arc::new(ttl_manager);
+
+        {
+            let fs_dir_lock = self.fs.fs_dir();
+            let mut fs_dir = fs_dir_lock.write();
+            let _ = fs_dir.init_ttl_manager(ttl_manager_arc.clone());
+        }
+
+        ttl_manager_arc.initialize()?;
+        self.start_ttl_heartbeat_checker(ttl_manager_arc)?;
+
+        info!("Inode ttl scheduler started successfully.");
+        Ok(())
+    }
+
+    fn start_ttl_heartbeat_checker(
+        &mut self,
+        ttl_manager: Arc<InodeTtlManager>,
+    ) -> CommonResult<()> {
+        info!("Starting inode ttl checker");
+        let ttl_config = TtlHeartbeatConfig {
+            task_name: "inode-ttl-checker".to_string(),
+            timeout_ms: self.fs.conf.ttl_checker_interval_ms * 2,
+        };
+        let heartbeat_checker = TtlHeartbeatChecker::new(ttl_manager, ttl_config);
+        let scheduler = ScheduledExecutor::new(
+            "inode-ttl-checker".to_string(),
+            self.fs.conf.ttl_checker_interval_ms,
+        );
+        scheduler.start(heartbeat_checker)?;
+        info!("Inode ttl checker started");
+        Ok(())
     }
 
     fn start_heartbeat_checker(
-        fs: MasterFilesystem,
+        fs: Arc<MasterFilesystem>,
```

Comment:
MasterFilesystem does not need to add Arc

Reply:
yes，i will fix it